vit-servicing-station-lib/src/db/models/funds.rs

```diff
@@ -3,41 +3,82 @@ use crate::db::{
     schema::funds,
     Db,
 };
+use crate::utils::datetime::unix_timestamp_to_datetime;
 use diesel::{ExpressionMethods, Insertable, Queryable};
 use serde::{Deserialize, Serialize};
 
 #[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+#[serde(into = "FundWithLegacyFields")]
 pub struct Fund {
     #[serde(default = "Default::default")]
     pub id: i32,
     #[serde(alias = "fundName")]
     pub fund_name: String,
     #[serde(alias = "fundGoal")]
     pub fund_goal: String,
-    #[serde(alias = "votingPowerInfo")]
-    pub voting_power_info: String,
     #[serde(alias = "votingPowerThreshold")]
     pub voting_power_threshold: i64,
     #[serde(alias = "rewardsInfo")]
     pub rewards_info: String,
     #[serde(alias = "fundStartTime")]
-    #[serde(serialize_with = "crate::utils::serde::serialize_unix_timestamp_as_rfc3339")]
     #[serde(deserialize_with = "crate::utils::serde::deserialize_unix_timestamp_from_rfc3339")]
     pub fund_start_time: i64,
     #[serde(alias = "fundEndTime")]
-    #[serde(serialize_with = "crate::utils::serde::serialize_unix_timestamp_as_rfc3339")]
     #[serde(deserialize_with = "crate::utils::serde::deserialize_unix_timestamp_from_rfc3339")]
     pub fund_end_time: i64,
     #[serde(alias = "nextFundStartTime")]
-    #[serde(serialize_with = "crate::utils::serde::serialize_unix_timestamp_as_rfc3339")]
     #[serde(deserialize_with = "crate::utils::serde::deserialize_unix_timestamp_from_rfc3339")]
     pub next_fund_start_time: i64,
+    #[serde(alias = "registrationSnapshotTime")]
+    #[serde(deserialize_with = "crate::utils::serde::deserialize_unix_timestamp_from_rfc3339")]
+    pub registration_snapshot_time: i64,
     #[serde(alias = "chainVotePlans", default = "Vec::new")]
     pub chain_vote_plans: Vec<Voteplan>,
     #[serde(default = "Vec::new")]
     pub challenges: Vec<Challenge>,
 }
 
+#[derive(Serialize)]
+struct FundWithLegacyFields {
```

Comment:
It is not clear to me, why do we separate this now?

Reply:
It's the easiest way I found to present the fields in JSON, while synthesizing the legacy field from the value of `registration_snapshot_time`.